validphys2/src/validphys/core.py

```diff
@@ -215,25 +216,37 @@ def get_members(self):
 
 
 class CommonDataSpec(TupleComp):
-    def __init__(self, datafile, sysfile, plotfiles, name=None, metadata=None):
+    def __init__(self, datafile, sysfile, plotfiles, name=None, metadata=None, legacy=True):
+        self.legacy = legacy
+        self._metadata = metadata
         self.datafile = datafile
         self.sysfile = sysfile
-        self.plotfiles = tuple(plotfiles)
-        self._name = name
-        self._metadata = metadata
-        super().__init__(datafile, sysfile, self.plotfiles)
+        if legacy:
+            self.plotfiles = tuple(plotfiles)
+            super().__init__(datafile, sysfile, self.plotfiles)
+        else:
+            self.plotfiles = False
+            super().__init__(name)
 
     @property
     def name(self):
         return self.metadata.name
 
     @property
     def nsys(self):
-        return self.metadata.nsys
+        if self.legacy:
+            return self.metadata.nsys
+        else:
+            cd = self.load()
+            return cd.nsys
 
     @property
     def ndata(self):
-        return self.metadata.ndata
+        if self.legacy:
+            return self.metadata.ndata
+        else:
+            cd = self.load()
+            return cd.nsys
```

Comment:
```suggestion
            return cd.ndata
```
shouldn't this be `ndata`?

Reply:
yes! Thanks!